solc/CommandLineInterface.cpp

```diff
@@ -377,8 +377,18 @@ void CommandLineInterface::handleGasEstimation(string const& _contract)
 	}
 }
 
-bool CommandLineInterface::readInputFilesAndConfigureFileReader()
+bool CommandLineInterface::readInputFiles()
 {
+	solAssert(!m_standardJsonInput.has_value(), "");
```

Comment:
I find `!x.has_value()` much more readable than `!x`. The latter often does not really read like a condition. And it makes `x` look like a pointer. I'd prefer to keep this as is.

Reply:
Yeah, one of these things where we'd need to agree on common style. Until then you probably will get corrected either way :)

BTW, I saw @ekpyron using `has_value()` in his PR so it's not just me :)

Reply:
Although I mainly use it, because people are telling me it's nicer and I don't really mind either way myself :-). Unfortunately though, as a consequence, I don't think I do it consistently either way myself these days :-).

Reply:
> because people are telling me it's nicer

Well, they're right :P

I'm not really sure why people are not bothered more by using non-boolean values in conditions but for me it just makes them harder to read so it's not worth saving a few keystrokes.

Reply:
I mainly use non-boolean values on conditions in branches, like ``if (auto someOptional = someFunction())``.
You could write ``if (auto someOptional = someFunction(); someOptional.has_value())`` for those, but in that case it seems like a pretty weird thing to do :-). In all other cases I could get used to using ``.has_value()``. But what do you propose for checking a pointer being not-``nullptr``? That's also a non-boolean value... we do have cases of ``!!ptr`` for that, but IMHO that doesn't really improve anything :-). And ``ptr != nullptr`` seems overly verbose.

Reply:
And if it's fine for a pointer to just implicitly have it convert to `bool`, I'm not sure why it should be wrong for an `optional`...

Reply:
I don't see anything wrong with `ptr != nullptr`. It reads like a full condition (if ptr is not null). `ptr` as a condition is just weird and I only use it because it's prevalent style in our code already. Otherwise I'd never do that.

I don't think it's wrong. I just think it's not as readable. This style is really common in Python and I really hate it.

> You could write `if (auto someOptional = someFunction(); someOptional.has_value())` for those, but in that case it seems like a pretty weird thing to do :-).

Agreed. I mean, I'd prefer to have has_value()` there but a preference for not having the second part in the `if` is much stronger :) It's too easy to overlook that part when skimming over code.

Reply:
Actually, e.g. in ``assert(ptr != nullptr)`` or even ``if(ptr != nullptr)`` I'd even be ok with it - I'm just extremely used to implicit pointer-to-bool conversions, so I'd probably have to be constantly reminded until I'm used to it :-D. But even more prevalent in the code is again the ``if (auto ptr = ...sth...)`` case, especially with the RTTI ``dynamic_cast`` cases... and there ``if (auto ptr = ...; ptr != nullptr)`` has the same problem as the optional case... but yeah, not sure. Maybe we should more broadly collect opinions. *Sometimes* using ``assert(!!ptr)``, *sometimes* ``assert(ptr)``, *sometimes* ``assert(someOptional)`` and *sometimes* ``assert(someOptional.has_value())`` randomly for sure isn't the nicest thing to do.